packages/loader/container-loader/src/container.ts

```diff
@@ -1329,6 +1303,13 @@ export class Container
 
 					await attachP;
 
+					assert(
+						isAttachedData(this.attachmentData),
+						"incorrect format in final attachment data",
+					);
+					this.containerStateManager.setSnapshot(
+						(this.attachmentData as AttachedData).snapshot,
```

Comment:
i would consider removing snapshot from attachmentData, and just have runRetriableAttachProcess return it. it was only on attachmentData so other places could access it, but now only the state manager needs it.